# Задача
Смоделировать игру против лотерейного автомата типа “777”. Игрок платит 1 руб., после чего выпадает случайное целое число, равномерно распределенное от 0 до 999. При некоторых значениях числа игрок получает выигрыш (см. справа)

777: 200 руб.

999: 100 руб.

555: 50 руб.

333: 15 руб.

111: 10 руб.

\*77: 5 руб.

\**7: 3 руб.

\*00: 2 руб.

\**0: 1 руб.

\* – любая цифра

Выгодна ли игра игроку? Сколько в среднем приобретает или теряет игрок за одну игру?

In [2]:
import numpy as np

In [55]:
trials = 1000000

entrance_fee = 1

prize_sum = 0

for i in range(trials):
    
    result = np.random.randint(0, 999)
    
    if result % 10 == 0: prize = 1 - entrance_fee
    if result % 100 == 0: prize = 2 - entrance_fee
    if result % 10 == 7: prize = 3 - entrance_fee
    if result % 100 == 77: prize = 5 - entrance_fee
    if result % 1000 == 111: prize = 10 - entrance_fee
    if result % 1000 == 333: prize = 15 - entrance_fee
    if result % 1000 == 555: prize = 50 - entrance_fee
    if result % 1000 == 999: prize = 100 - entrance_fee
    if result % 1000 == 777: prize = 200 - entrance_fee
    else: prize = 0 - entrance_fee
    
    prize_sum = prize_sum + prize
        
print('В среднем за одну игрок получает:', prize_sum / trials, 'рубль(ей)')

В среднем за одну игрок получает: -0.81 рубль(ей)


# Ответ
Если сыграть в игру миллион раз при том, что автомат генерирует числа действительно случайным образом, то игрок будет в среднем выигрывать 0.81 рубля. Выгодно это или нет - решать ему.